spec/models/authenticator/ldap_spec.rb

```diff
@@ -163,6 +164,19 @@ def authenticate
     let(:username) { 'alice' }
     let(:password) { 'secret' }
 
+    context "when using LDAP" do
+      let(:config) { super().merge(:ldap_role => true) }
+
+      it "encrypts password for queuing" do
+        called_encrypt_ldap_password = false
+        allow(MiqTask).to receive(:create).and_return(double(:id => :return_value))
+        allow(MiqQueue).to receive(:put) { :return_value }
+        allow(subject).to receive(:encrypt_ldap_password) { called_encrypt_ldap_password = true }
+        authenticate
+        expect(called_encrypt_ldap_password).to be_truthy
```

Comment:
I see you mean `:return_value` as a dummy value, derp. In this case I'd personally use an Integer on line 172 (whatever, though) but more importantly you needn't give line 173 a block, `allow(MiqQueue).to receive(:put)` will work just fine :ok_hand: 

I suggest the following:

```
    context "when using LDAP" do
      let(:config) { super().merge(:ldap_role => true) }

      before do
        allow(MiqQueue).to receive(:put)
      end

      it "encrypts password for queuing" do
        expect(subject).to receive(:authorize_queue)
        authenticate
      end
    end
```

Whatcha think?